modules/xmpp/xmpp.js

```diff
@@ -490,50 +496,62 @@ export default class XMPP extends Listenable {
     /**
      * Disconnects this from the XMPP server (if this is connected).
      *
-     * @param ev optionally, the event which triggered the necessity to
+     * @param {Object} ev - Optionally, the event which triggered the necessity to
      * disconnect from the XMPP server (e.g. beforeunload, unload).
+     * @returns {Promise} - Resoves when the disconnect process is finished or rejects with an error.
      */
     disconnect(ev) {
         if (this.disconnectInProgress || !this.connection) {
             this.eventEmitter.emit(JitsiConnectionEvents.WRONG_STATE);
 
-            return;
+            return Promise.reject(new Error('Wrong connection state!'));
```

Comment:
Would it be useful to add information about there being a disconnect in progress or there isn't a connection?

Reply:
hmm... in jitsi-meet we don't care...

Actually I don't see benefit because I imagine in both cases you would just ignore this.